subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r81/DependencyArtifactDownloadProgressEventCrossVersionTest.groovy

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r81
+
+import org.gradle.integtests.tooling.fixture.AbstractHttpCrossVersionSpec
+import org.gradle.integtests.tooling.fixture.ProgressEvents
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.test.fixtures.Flaky
+import org.gradle.tooling.ProjectConnection
+import org.gradle.tooling.events.OperationType
+import org.gradle.tooling.events.download.internal.NotFoundFileDownloadSuccessResult
+
+@ToolingApiVersion(">=8.1")
```

Comment:
What happens if an older Tooling API version is used?

Reply:
good question, you should get a `DefaultFileDownloadSuccessResult` because the `NotFoundFileDownloadSuccessResult` is not in the tooling-api?

Reply:
This is to ensure forward-compatibility for older Tooling API releases.